### PR TITLE
Switch EKS-A Admin image to AL2 from Ubuntu

### PR DESCRIPTION
**Pull request by abhay-krishna**
This PR changes the EKS-A Admin image for Snow to be built from AL2 instead of Ubuntu.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: projects/aws/eks-a-admin-image/provisioners/al2/download_eksa_artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
Comment (g-gaston) [inline projects/aws/eks-a-admin-image/provisioners/al2/download_eksa_artifacts.sh:1]: does this file need to be specific for al2? I would expect this kind of eks-a commands to be OS agnostic

File: projects/aws/eks-a-admin-image/provisioners/al2/install_eksa.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+set -o pipefail
+
+# YQ_URL needs to be provided to the script
+
+EKSCTL_VERSION="${EKSCTL_VERSION:-latest}" # The eksctl version to install (example v0.0.0)
+EKSA_RELEASE_MANIFEST_URL="${EKSA_RELEASE_MANIFEST_URL:-https://anywhere-assets.eks.amazonaws.com/releases/eks-a/manifest.yaml}" # The url pointing to the eks-a releases manifest
+EKSA_VERSION="${EKSA_VERSION:-latest}" # The eks-a version to install (example v0.0.0)
+
+# Install depencies
+sudo yum install -y tar
```
Comment (g-gaston) [inline projects/aws/eks-a-admin-image/provisioners/al2/install_eksa.sh:17]: maybe we could move this to a separate script so we can share the rest of the code between OS's?

File: projects/aws/eks-a-admin-image/provisioners/al2/install_kubectl.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
```
Comment (g-gaston) [inline projects/aws/eks-a-admin-image/provisioners/al2/install_kubectl.sh:1]: this one doesn't seem al2 specific?

**Review by jiayiwang7 — APPROVED** [review]

File: projects/aws/eks-a-admin-image/provisioners/al2/install_kubectl.sh (hunk 3 above)
Comment (abhay-krishna) [reply projects/aws/eks-a-admin-image/provisioners/al2/install_kubectl.sh]: Using the common script now.

File: projects/aws/eks-a-admin-image/provisioners/download_eksa_artifacts.sh [hunk 4]
```diff
@@ -9,10 +9,12 @@ EKSA_MANIFESTS_PATH="${EKSA_MANIFESTS_PATH:-/usr/lib/eks-a/manifests}" # The ful
...
+sudo -E env "PATH=$PATH" eksctl anywhere download artifacts --retain-dir --download-dir $EKSA_MANIFESTS_PATH -v4
 sudo chmod -R a+r $EKSA_MANIFESTS_PATH
 
 ARTIFACTS_NAME_DIR=$(dirname $EKSA_ARTIFACTS_TAR_PATH)
 sudo mkdir -p $ARTIFACTS_NAME_DIR
-sudo eksctl anywhere download images -o $EKSA_ARTIFACTS_TAR_PATH -v4
+sudo -E env "PATH=$PATH" eksctl anywhere download images -o $EKSA_ARTIFACTS_TAR_PATH -v4
 sudo chmod -R a+r $ARTIFACTS_NAME_DIR
+
+sudo docker system prune --volumes --force
```
Comment (g-gaston) [inline projects/aws/eks-a-admin-image/provisioners/download_eksa_artifacts.sh:20]: super nit: add a comment to explain why we do this pls pls

File: projects/aws/eks-a-admin-image/provisioners/install_eksa.sh [hunk 5]
```diff
@@ -38,4 +32,4 @@ EKSA_TAR_URL=$(curl -L --silent $EKSA_RELEASE_MANIFEST_URL | yq e '.spec.release
 curl -L --silent $EKSA_TAR_URL | tar xz -C ./eksa_tmp
 
 sudo mv eksa_tmp/eksctl-anywhere /usr/local/bin/
-rm -rf eksa_tmpq
+rm -rf eksa_tmp
```
Comment (g-gaston) [inline projects/aws/eks-a-admin-image/provisioners/install_eksa.sh:35]: Good catch :)

**Review by g-gaston — APPROVED** [review]

File: projects/aws/eks-a-admin-image/provisioners/download_eksa_artifacts.sh (hunk 4 above)
Comment (abhay-krishna) [reply projects/aws/eks-a-admin-image/provisioners/download_eksa_artifacts.sh:26]: Done!

**Pull request merged**